app/src/androidTest/java/br/com/catbag/gifreduxsample/ui/giflist/GifListActivityTest.java

```diff
@@ -2,154 +2,338 @@
+        dispatchFakeReduceAction(createStateFromGifs(createFakeGifList(gifListSize)));
+
+        mActivityTestRule.launchActivity(new Intent());
+
+        FeedComponent feed = (FeedComponent) getActivity().findViewById(R.id.feed);
+        RecyclerView recyclerView = (RecyclerView) feed.getChildAt(0);
+        recyclerView.setId(View.generateViewId());
+
+        UiTestLocker locker = new UiTestLocker(getActivity());
+
+        onView(withId(recyclerView.getId()))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(gifListSize-1, click()));
 
-        assert(!mActivityTestRule.getActivity().getGifWrapper().getDrawable().isRunning());
+        locker.registerIdlingResource();
 
+        int lastViewId = View.generateViewId();
+        FrameLayout frame = ((FrameLayout)recyclerView.getChildAt(recyclerView.getChildCount()-1));
+        View lastView = frame.findViewById(R.id.gif_image);
+        lastView.setId(lastViewId);
+
+        onView(withId(lastViewId))
+                .check(matches(withPlayingGifDrawable()));
+
+        lastView.setId(R.id.gif_image);
+        locker.unregisterIdlingResource();
     }
 
     @Test
-    public void gifDownloadFailedTest() {
-        mockFileDownloaderToAlwaysFail();
+    public void whenScrollUpAndPlayFirstItemTest() {
+        int gifListSize = 5;
+        dispatchFakeReduceAction(createStateFromGifs(createFakeGifList(gifListSize)));
 
         mActivityTestRule.launchActivity(new Intent());
 
-        onView(withText("Download error")).inRoot(withToast())
-                .check(matches(isDisplayed()));
+        FeedComponent feed = (FeedComponent) getActivity().findViewById(R.id.feed);
+        RecyclerView recyclerView = (RecyclerView) feed.getChildAt(0);
+        recyclerView.setId(View.generateViewId());
+
+        UiTestLocker locker = new UiTestLocker(getActivity());
+
+        onView(withId(recyclerView.getId())).perform(scrollToPosition(gifListSize-1));
+        onView(withId(recyclerView.getId()))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+
+        locker.registerIdlingResource();
+
+        int firstViewId = View.generateViewId();
+        FrameLayout frame = ((FrameLayout)recyclerView.getChildAt(0));
+        View firstView = frame.findViewById(R.id.gif_image);
+        firstView.setId(firstViewId);
+
+        onView(withId(firstViewId))
+                .check(matches(withPlayingGifDrawable()));
+
+        firstView.setId(R.id.gif_image);
+        locker.unregisterIdlingResource();
+    }
+
+    private void dispatchFakeReduceAction(AppState state) {
+        getFluxxan().getDispatcher()
+                .dispatch(new Action(FakeReducer.FAKE_REDUCE_ACTION, state));
+    }
+
+    private GifListActivity getActivity(){
+        return mActivityTestRule.getActivity();
+    }
+
+    private Fluxxan<AppState> getFluxxan(){
+        return MyApp.getFluxxan();
+    }
+
+        private List<Gif> createFakeGifList(int size) {
```

Comment:
Check, but with lambda checkstyle has some problems...i will open a issue to solve this problem with more energy on future...see this: https://github.com/checkstyle/checkstyle/issues/281